### ISSUE TITLE
Setting custom chromeDriver path leaves rogue chromedriver processes

### ISSUE THREAD

**Issue by hackel** [laravel/dusk#370]
There is a recurrance of issue #341 when specifying a custom chromeDriver path.  I am using `static::useChromedriver(base_path('vendor/bin/chromedriver')` before startChromeDriver in my DuskTestCase::prepare method.

For some reason, the Symfony Process::start method only adds 'exec' to the command if commandline is an array.  When a custom chromeriver is set, the Process class is created using the ProcessBuilder instead of instantiating it directly, which sets commandline to a string instead of an array, and thus 'exec' never gets prepended.  (See \Laravel\Dusk\Chrome\ChromeProcess::toProcess)

I'm not really clear on why they are different.  Unless there is something I'm missing, both process() and fromProcessBuilder() do the same thing.

**Comment by deleugpn** [comment]
I'll try to answer all your doubts in snippets

1- Dusk uses ProcessBuilder to support Windows. Historically, Dusk used to start every process through ProcessBuilder. In order to improve Linux and Mac support, the direct usage of Process was introduced in a ChromeProcess refactoring.

2- The reason why custom driver uses ProcessBuilder instead of Process is to keep backward compatibility during the refactoring.

Basically if it's custom driver, we kept the old behaviour of using PB. If it's Windows we use PB. If it's Mac and Linux we use Process.

**Comment by hackel** [comment]
Until this is fixed (either in Dusk or Symfony Process), I was able to work around this by overriding startChromeDriver:
```php
    public static function startChromeDriver()
    {
        static::$chromeProcess = static::buildChromeProcess();

        static::$chromeProcess->setCommandLine('exec ' . static::$chromeProcess->getCommandLine());

        static::$chromeProcess->start();

        static::afterClass(function () {
            static::stopChromeDriver();
        });
    }
```

This issue arose again due to a Chromium upgrade on my desktop, requiring chromedriver 2.33.  It would be nice if Dusk depended on an external chromedriver package such as lbaey/chromedriver that can be upgraded more frequently than Dusk itself.

**Comment by driesvints** [comment]
Closing this issue because it's already solved, old or not relevant anymore. Feel free to reply if you're still experiencing this issue.
